src/Compilers/VisualBasic/Portable/Symbols/Tuples/TupleFieldSymbol.vb

```diff
@@ -0,0 +1,185 @@
+            Get
+                Return Symbol.GetDeclaringSyntaxReferenceHelper(Of VisualBasicSyntaxNode)(Me._locations)
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property IsImplicitlyDeclared() As Boolean
+            Get
+                Return False
+            End Get
+        End Property
+
+        Friend Overrides ReadOnly Property TypeLayoutOffset() As Integer?
+            Get
+                Dim flag As Boolean = Me._underlyingField.ContainingType IsNot Me._containingTuple.TupleUnderlyingType
+                Dim result As Integer?
+                If flag Then
+                    result = Nothing
+                Else
+                    result = MyBase.TypeLayoutOffset
+                End If
+                Return result
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property AssociatedSymbol() As Symbol
+            Get
+                Dim flag As Boolean = Me._underlyingField.ContainingType IsNot Me._containingTuple.TupleUnderlyingType
+                Dim result As Symbol
+                If flag Then
+                    result = Nothing
+                Else
+                    result = MyBase.AssociatedSymbol
+                End If
+                Return result
+            End Get
+        End Property
+
+        Public Sub New(container As TupleTypeSymbol, underlyingField As FieldSymbol, tupleFieldId As Integer, location As Location)
+            MyBase.New(container, underlyingField, tupleFieldId)
+            Me._locations = (If((location Is Nothing), ImmutableArray(Of Location).Empty, ImmutableArray.Create(Of Location)(location)))
+        End Sub
+    End Class
+
+    ''' <summary>
+    ''' Represents an element field of a tuple type (such as (int a, byte b).a, or (int a, byte b).b)
+    ''' that is backed by a real field with a different name within the tuple underlying type.
+    ''' </summary>
+    Friend Class TupleRenamedElementFieldSymbol
+        Inherits TupleElementFieldSymbol
+
+        Private _name As String
+
+        Public Overrides ReadOnly Property Name() As String
+            Get
+                Return Me._name
+            End Get
+        End Property
+
+        Friend Overrides ReadOnly Property TypeLayoutOffset() As Integer?
+            Get
+                Return Nothing
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property AssociatedSymbol() As Symbol
+            Get
+                Return Nothing
+            End Get
+        End Property
+
+        Public Sub New(container As TupleTypeSymbol, underlyingField As FieldSymbol, name As String, tupleElementOrdinal As Integer, location As Location)
+            MyBase.New(container, underlyingField, tupleElementOrdinal, location)
+            Debug.Assert(name <> Nothing)
+            Debug.Assert(name <> underlyingField.Name)
+            Me._name = name
+        End Sub
```

Comment:
These derived field types do not override `Equals` which means `New TupleFieldSymbol(container, underlyingField) = New TupleRenamedElementFieldSymbol(container, underlyingField, "name")`. Is that expected?